server/src/routes/api/user.mjs

```diff
@@ -27,21 +27,49 @@ import * as stateManagement from '../../stateManagement.mjs';
 
 // --- Route Handlers ---
 
-// POST /api/v1/user
+// PUT /api/v1/user/:userId?
+// POST /api/v1/user/:userId?
 // Expected body: N/A
+// URL Param "userId" - Optional
 function addUser(req, res) {
-  // Generate a unique userId for this user (no vanity userIds, at least for now)
-  const userId = uuidv4();
+  let userId;
+  if (req.params.userId){
```

Comment:
It feels like this function could be refactored to avoid dup code. Just something up front that either uses the given user id if given or generates one. Then the core logic that does the stateManagement.addUser, userManagement.addUser, and returns success or error.